### PR TITLE
Fix dnd with chrome

### LINKED ISSUE #49: DND in Chrome not dropping or hovering
This doesn't work at the moment - a new window for the tab you start dragging is created.

### PR DESCRIPTION
**Pull request by okratitan**
This fixes dnd with chrome and fixes #49

### CODE REVIEW

**Comment by okratitan** [comment]
Before approving we should discuss if reversing the stack list is correct - or if we should use the bottom to top order in stack list to begin with.  I didn't make this change yet because I couldn't remember if there a reason we went to top to bottom on stack list.

**Comment by andydotxyz** [comment]
Great catch. As for the ordering I am not sure. I think the window order was based on the order we would display them on switcher. So either we reverse it here, or in the switcher code. We should probably document the order in the `Stack` interface I guess.

**Review by andydotxyz — DISMISSED** [review]

File: wm.go [hunk 1]
```diff
@@ -14,13 +14,13 @@ type WindowManager interface {
 }
 
 // Stack describes an ordered list of windows.
-// The order of the windows in this list matches the stacking order on screen.
-// TopWindow() returns the 0th element with each item after that being stacked below the previous.
```
Comment (andydotxyz) [inline wm.go:18]: Wow, I forgot that we defined it this way around.
Are we certain that reversing it for all platforms is the right thing to do?

**Comment by andydotxyz** [comment]
I jumped over to WinGo as an example of another WM in go - they have stacked it how we previously did:

https://github.com/BurntSushi/wingo/blob/30b336cbb88d32c12c14768d5ff89b3d55560081/stack/stack.go

**Comment by okratitan** [comment]
Yes and they had to provide a helper function to convert it.  Walkinh the slice to convert the stack changes on every focus, raise, lower, add, and remove of windows surely isn't ideal, and appending new windows or raised windows to the end of a slice surely makes more sense than inserting them at the beginning.

File: wm.go (hunk 1 above)
Comment (okratitan) [reply wm.go:18]: I can't see any reason why it wouldn't be fine for any other platform -- I'm not certain though.  However I'm less certain that having it top to bottom is the right thing to do.  So far we have one implementation - X and it says bottom to top.  That should be fine for Wayland from what I can tell so far too...  I would say between the two there isn't a lot of certainty of which one is better - but the one concrete thing we have and know is how the one platform we have wants it bottom to top and from a code standup that means less conversions and slice walking so I think that tips that scales for me.

**Comment by andydotxyz** [comment]
If The X11 driver code wants to use the reverse order for performance reasons then that is fair enough.
But that in itself does not indicate that we should reverse the exposed API...

Another approach would be that the `Windows` method could reverse it so that the backwards list is internal to the driver only and our API remains unaltered.

**Comment by okratitan** [comment]
Fair enough - reverted the interface/switcher changes and reverse the stack on Windows() method

**Review by andydotxyz — APPROVED** [review]
Thanks, this feels like a nice clean change and a very welcome fix :)

**Pull request merged**